devtools/forcefield-scripts/amber/amber2omm.py

```diff
@@ -0,0 +1,844 @@
+                                       % leaprc_name)
+        if MODE == 'LEAPRC':
+            ffxml_name = convert_leaprc(files, ffxml_dir=ffxml_dir, ignore=ignore,
+                         provenance=provenance, write_unused=write_unused,
+                         filter_warnings=filter_warnings, split_filename=True)
+        elif MODE == 'RECIPE':
+            ffxml_name = convert_recipe(files, solvent_file=solvent_file,
+                         ffxml_dir=ffxml_dir, provenance=provenance,
+                         ffxml_basename=recipe_name)
+        if verbose: print('Validating the conversion...')
+        tested = False
+        for test in leaprc_test:
+            if test == 'protein':
+                validate_protein(ffxml_name, leaprc_name)
+                tested = True
+            elif test == 'nucleic':
+                validate_nucleic(ffxml_name, leaprc_name)
+                tested = True
+            elif test == 'protein_ua':
+                validate_protein(ffxml_name, leaprc_name, united_atom=True)
+                tested = True
+            elif test == 'protein_phospho':
+                validate_phospho_protein(ffxml_name, leaprc_name)
+                tested = True
+            elif test == 'gaff':
+                validate_gaff(ffxml_name, leaprc_name)
+                tested = True
+            elif test == 'water_ion':
+                validate_water_ion(ffxml_name, files, solvent_name, recipe_name,
+                standard_ffxml=standard_ffxml)
+                tested = True
+        if not tested:
+            raise RuntimeError('No validation tests have been run for %s' %
+                                leaprc_name)
+
+def assert_energies(prmtop, inpcrd, ffxml, system_name='unknown', tolerance=1e-5,
+                    improper_tolerance=1e-2, units=u.kilojoules_per_mole):
+    # AMBER
+    parm_amber = parmed.load_file(prmtop, inpcrd)
+    system_amber = parm_amber.createSystem(splitDihedrals=True)
+    amber_energies = parmed.openmm.energy_decomposition_system(parm_amber,
+                     system_amber, nrg=units)
+    # OpenMM-ffxml
+    if isinstance(ffxml, str):
+        ff = app.ForceField(ffxml)
+    else:
+        ff = app.ForceField(*ffxml)
+    system_omm = ff.createSystem(parm_amber.topology)
+    parm_omm = parmed.openmm.load_topology(parm_amber.topology, system_omm,
+                                           xyz=parm_amber.positions)
+    system_omm = parm_omm.createSystem(splitDihedrals=True)
+    omm_energies = parmed.openmm.energy_decomposition_system(parm_omm,
+                   system_omm, nrg=units)
+
+    # calc rel energies and assert
+    rel_energies = []
+    for i, j in zip(amber_energies, omm_energies):
+        if i[0] != j[0]:
+            raise RuntimeError('Mismatch in energy tuples naming.')
+        if i[1] != 0:
+            rel_energies.append((i[0], (abs(i[1]-j[1])/abs(i[1]))))
+        else:
+            if j[1] != 0:
```

Comment:
Change direct comparisons to 0 to be > small number.  In several of my tests with OpenMM/ParmEd/Amber I've seen energies ~1e-300, which would fail this test but clearly still be OK.